### PR TITLE
feat(pollux): add anoncreds prooving implementation

### PR DESCRIPTION
**Pull request by goncalo-frade-iohk**
# Overview
This will add implementation so we can create presentation with a anoncred credential.

Fixes #ATL-5203

### My PR contains...
* [ ] No code changes (changes to documentation, CI, metadata, etc.)
* [ ] Bug fixes (non-breaking change which fixes an issue)
* [ ] Improvements (misc. changes to existing features)
* [x] Features (non-breaking change which adds functionality)

### My changes...
* [ ] are breaking changes
* [x] are not breaking changes
* [ ] If yes to the above: I have updated the documentation accordingly

### Documentation
* [ ] My changes do not require a change to the project documentation
* [ ] My changes require a change to the project documentation
* [x] If yes to the above: I have updated the documentation accordingly

### Tests
* [ ] My changes can not or do not need to be tested
* [ ] My changes can and should be tested by unit and/or integration tests
* [x] If yes to the above: I have added tests to cover my changes
* [ ] If yes to the above: I have taken care to cover edge cases in my tests

### CODE REVIEW

**Review by hamada147 — APPROVED** [review]

File: AtalaPrismSDK/Pollux/Sources/Models/AnonCreds/AnoncredsCredentialStack+ProvableCredential.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        {
+            return try AnoncredsPresentation().createPresentation(
+                stack: self,
+                request: requestStr,
+                linkSecret: linkSecret,
+                attributes: attributes,
+                predicates: predicates
+            )
+        } else {
+            return try AnoncredsPresentation().createPresentation(
```
Comment (elribonazo) [inline AtalaPrismSDK/Pollux/Sources/Models/AnonCreds/AnoncredsCredentialStack+ProvableCredential.swift:50]: Small comment or suggestion but is this if condition or else really needed? 
computeAttributes and computePredicates will not process much if the arrays or object with attributes or predicates are empty right?

Could be wrong :)

Comment (goncalo-frade-iohk) [reply AtalaPrismSDK/Pollux/Sources/Models/AnonCreds/AnoncredsCredentialStack+ProvableCredential.swift:50]: Yeah so the way I envision this logic was.

If the user provides ZKP Parameters then we don't do anything, if they wish to provide ZKP parameters with no attributes or predicates, its their option, so we don't compute them.

If the user doesn't provide ZKP parameters then we compute a "default" version based on the presentation request.
